dmref_analyzer/FileBrowser.py

```diff
@@ -1,55 +1,62 @@
-import glob, os
+from _typeshed import StrOrBytesPath
+import glob
+import os
 import fnmatch
 import re
 import numpy as np
 import util
 
 import pandas as pd
 import sys
+from os.path import dirname, abspath
 
 
 class FileBrowser(object):
     def __init__(self, root_dir=None, param_rule_file=None):
         if root_dir is None:
-            self.root_dir = os.path.expanduser('~/Dropbox/DMREF/Database/')
+            self.root_dir = dirname(dirname(abspath(__file__)))
```

Comment:
This won't work, it used to be the "Database" directory, you changed it to become

dmref_analyzer/FileBrowser.py/../../

Doesn't this become dmref_analyzer/../ ?

Reply:
I don't think so. I just tested this part of code on my system and it printed below directory:
/home/ashkan/Desktop/project/dmref_analyzer

dmref_analyzer is the main repo directory.

abspath gives "/home/ashkan/Desktop/project/dmref_analyzer/dmref_analyzer/FileBrowser.py". Calling dirname function in the first time gives >>> "/home/ashkan/Desktop/project/dmref_analyzer/dmref_analyzer/"
and calling it the second time gives >>> "/home/ashkan/Desktop/project/dmref_analyzer/"
which is the repo directory on any system.
Am I correct?

Reply:
/home/ashkan/Desktop/project/dmref_analyzer indeed is what I said earlier: dmref_analyzer/../, i.e. the parent directory of dmref_analyzer. Note that dmref_analyzer here is NOT the main repo directory, but the "module" directory, its parent directory, on your system, is /home/ashkan/Desktop/project/dmref_analyzer, but on a different system, it could be /home/xiangyu/project/dmref_repo. 
But yea, it is the repo's root directory. 

However, see line 14 in the old code, the path is "~/Dropbox/DMREF/Database/". If you check the shared Dropbox directory, you can see that dir contains several names (Ji Hao, Sanwei, etc.). Which is not the same as the root directory for the repo. 

To test if all the paths work, could you try making another change?
Go to __init__.py and change the line from
from DataMatrix import DataMatrix
to
from .DataMatrix import DataMatrix

Then try running the genDendro file and see what errors you run into.

Reply:
I just ran the code again, and I got the same the error about no module named dmref_analyzer.